configure.ac

```diff
@@ -83,69 +83,20 @@ AC_TYPE_UINT16_T
 AC_TYPE_UINT32_T
 AC_C_INLINE
 AC_FUNC_STRERROR_R
-
-# https://lists.gnu.org/archive/html/autoconf-commit/2012-12/msg00004.html
-# AC_C__GENERIC
-# -------------
-# Define HAVE_C__GENERIC if _Generic works, a la C11.
-AN_IDENTIFIER([_Generic], [AC_C__GENERIC])
-AC_DEFUN([AC_C__GENERIC],
-[AC_CACHE_CHECK([for _Generic], ac_cv_c__Generic,
-[AC_COMPILE_IFELSE(
-   [AC_LANG_SOURCE(
-      [[int
-        main (int argc, char **argv)
-        {
-          int a = _Generic (argc, int: argc = 1);
-          int *b = &_Generic (argc, default: argc);
-          char ***c = _Generic (argv, int: argc, default: argv ? &argv : 0);
-          _Generic (1 ? 0 : b, int: a, default: b) = &argc;
-          _Generic (a = 1, default: a) = 3;
-          return a + !b + !c;
-        }
-      ]])],
-   [ac_cv_c__Generic=yes],
-   [ac_cv_c__Generic=no])])
-if test $ac_cv_c__Generic = yes; then
-  AC_DEFINE([HAVE_C__GENERIC], 1,
-           [Define to 1 if C11-style _Generic works.])
-fi
-])# AC_C__GENERIC
-
 AC_C__GENERIC
-
-# AC_C___ATOMIC
-# -------------
-# Define HAVE_C___ATOMIC if __atomic works.
-AN_IDENTIFIER([__atomic], [AC_C___ATOMIC])
-AC_DEFUN([AC_C___ATOMIC],
-[AC_CACHE_CHECK([for __atomic], ac_cv_c___atomic,
-[AC_COMPILE_IFELSE(
-   [AC_LANG_SOURCE(
-      [[int
-        main (int argc, char **argv)
-        {
-          volatile unsigned long ul1 = 1, ul2 = 0, ul3 = 2;
-          __atomic_compare_exchange(&ul1, &ul2, &ul3, 1, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
-          __atomic_fetch_add(&ul1, 1, __ATOMIC_SEQ_CST);
-          __atomic_fetch_sub(&ul3, 1, __ATOMIC_SEQ_CST);
-          volatile unsigned long long ull1 = 1, ull2 = 0, ull3 = 2;
-          __atomic_compare_exchange(&ull1, &ull2, &ull3, 1, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
-          __atomic_fetch_add(&ull1, 1, __ATOMIC_SEQ_CST);
-          __atomic_fetch_sub(&ull3, 1, __ATOMIC_SEQ_CST);
-          return 0;
-        }
-      ]])],
-   [ac_cv_c___atomic=yes],
-   [ac_cv_c___atomic=no])])
-if test $ac_cv_c___atomic = yes; then
-  AC_DEFINE([HAVE_C___ATOMIC], 1,
-           [Define to 1 if __atomic operations work.])
-fi
-])# AC_C___ATOMIC
-
 AC_C___ATOMIC
+AC_CHECK_SIZEOF([void *])
+AC_CANONICAL_HOST
 
+AC_ARG_VAR([SSE_CANDIDATE], [C compiler flags for SSE])
+AS_CASE([$host_cpu],
+	[i?86], [SSE_CANDIDATE="yes"]
+)
+AC_SUBST([SSE_CANDIDATE])
+if test "${SSE_CANDIDATE}" = "yes"; then
+	opt="-msse2 -mfpmath=sse"
+	AX_CHECK_COMPILE_FLAG($opt, [CFLAGS="$CFLAGS $opt"], [])
+fi
```

Comment:
please avoid auto setting CFLAGS for optimizations as you cannot really know it this is supported or needed. the fact that the compiler supports a flag does not mean that the runtime environment does.

you must let packager set the CFLAGS of optimizations, and let it be.

if you insist to add CFLAGS you must use --enable-optimization or similar parameter that if set off you avoid messing up with flags.